src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -292,14 +286,17 @@ private GeneratorOptions getGeneratorOptions() {
     optionsBuilder.mapSize((int) (mapSizeSpinner.getValue() * KM_TO_PIXEL_FACTOR));
     optionsBuilder.numTeams(numTeamsSpinner.getValue());
     optionsBuilder.generationType(generationTypeComboBox.getValue());
+    optionsBuilder.symmetry(symmetryComboBox.getValue());
     optionsBuilder.style(mapStyleComboBox.getValue());
-    optionsBuilder.biome(biomeComboBox.getValue());
-    getSliderValue(waterSlider, waterRandom).ifPresent(value -> optionsBuilder.landDensity(1 - value));
-    getSliderValue(plateauSlider, plateauRandom).ifPresent(optionsBuilder::plateauDensity);
-    getSliderValue(mountainSlider, mountainRandom).ifPresent(optionsBuilder::mountainDensity);
-    getSliderValue(rampSlider, rampRandom).ifPresent(optionsBuilder::rampDensity);
-    getSliderValue(mexSlider, mexRandom).ifPresent(optionsBuilder::mexDensity);
-    getSliderValue(reclaimSlider, reclaimRandom).ifPresent(optionsBuilder::reclaimDensity);
+    if (fixedSeedCheckBox.isSelected() && generationTypeComboBox.getValue() == GenerationType.CASUAL) {
+      optionsBuilder.seed(seedTextField.getText());
+    }
+    if (customStyleCheckBox.isSelected() && generationTypeComboBox.getValue() == GenerationType.CASUAL) {
+      optionsBuilder.terrainGenerator(terrainComboBox.getValue());
+      optionsBuilder.textureGenerator(biomeComboBox.getValue());
+      optionsBuilder.resourceGenerator(resourcesComboBox.getValue());
+      optionsBuilder.propGenerator(propsComboBox.getValue());
+    }
```

Comment:
This can set the mapstyle and custom style options which is an error in the generator